test/semanticValidatorTests.ts

```
@@ -8,51 +8,205 @@ import * as validate from "../lib/validate";
+      const specPath = `${testPath}/semanticValidation/specification/validateOperation/MULTIPLE_BODY_PARAMETERS.json`;
+      const result = await validate.validateSpec(specPath, undefined);
+      assert(result.validityStatus === false);
+      assert.strictEqual(result.validateSpec?.errors?.[0].code, "MULTIPLE_BODY_PARAMETERS");
+    });
+    // INVALID_PARAMETER_COMBINATION
+    it("should fail when operation has a body parameter and a formData parameter", async () => {
+      const specPath = `${testPath}/semanticValidation/specification/validateOperation/INVALID_PARAMETER_COMBINATION.json`;
+      const result = await validate.validateSpec(specPath, undefined);
+      assert(result.validityStatus === false);
+      assert.strictEqual(result.validateSpec?.errors?.[1].code, "INVALID_PARAMETER_COMBINATION");
+    });
+    // MISSING_PATH_PARAMETER_DECLARATION
+    it("should fail when path parameter is defined but is not declared", async () => {
+      const specPath = `${testPath}/semanticValidation/specification/validateOperation/MISSING_PATH_PARAMETER_DECLARATION.json`;
+      const result = await validate.validateSpec(specPath, undefined);
+      assert(result.validityStatus === false);
+      assert.strictEqual(
+        result.validateSpec?.errors?.[0].code,
+        "MISSING_PATH_PARAMETER_DECLARATION"
+      );
+    });
+    // MISSING_PATH_PARAMETER_DEFINITION
+    it("should fail when path parameter is declared but is not defined", async () => {
+      const specPath = `${testPath}/semanticValidation/specification/validateOperation/MISSING_PATH_PARAMETER_DEFINITION.json`;
+      const result = await validate.validateSpec(specPath, undefined);
+      assert(result.validityStatus === false);
+      assert.strictEqual(
+        result.validateSpec?.errors?.[0].code,
+        "MISSING_PATH_PARAMETER_DEFINITION"
+      );
+    });
   });
 });
+
+// it("should validate correctly when the spec contains an x-ms-parameterized-host", async () => {
```

Are all of below tests not valid?

---

just the third test("should succeed when discriminator is not a required property and the error is suppressed") is not valid

these are the original test codes
should I put back the ones that work?